src/components/cartCard/cartCard.js

```diff
@@ -13,7 +12,61 @@ export default function CartCard({ cartitem }) {
             }
         });
         setUser({ ...user, cart: res.data.cart });
+    }
+    const moveToWishlist = async () => {
+        if (user.wishlist.length > 0) {
+            if (!user.wishlist.some(item => item._id === _id)) {
+                //if not in wishlist
+                const res = await axios.post("/api/user/wishlist", { product }, {
+                    headers: {
+                        authorization: encodedToken
+                    }
+                });
+                console.log(res.data, "...", product);
+                setUser({ ...user, wishlist: res.data.wishlist });
+                handleRemove();
+            }
+        } else {
+            const res = await axios.post("/api/user/wishlist", { product }, {
+                headers: {
+                    authorization: encodedToken
+                }
+            });
+            console.log(res.data, "...", product);
+            setUser({ ...user, wishlist: res.data.wishlist });
+            handleRemove();
+        }
+
+        // setUser({ ...user, wishlist: res.data.wishlist });
+        console.log(user.wishlist);
+
+    }
+    const changeQty = async (e) => {
+        if (e === "+") {
+            const res = await axios.post(`/api/user/cart/${_id}`, {
+                action: {
+                    type: "increment"
+                }
+            }, {
+                    headers: {
+                        authorization: encodedToken
+                    }
+                }
+            )
+        }
+        else {
+            const res = await axios.post(`/api/user/cart/${_id}`, {
+                action: {
+                    type: "decrement"
+                }
+            }, {
+                    headers: {
+                        authorization: encodedToken
+                    }
```

Comment:
suggestion:You dont need to call Api 2 times can use ternery operator inside action to check if value of e is + or - based on that u can pass type to action
example:
action:{
  type:e==="+"?"increment":"decrement"
}

Reply:
Done, thanks for mentioning this.